client/src/modules/fill_form/fill_form.service.js

```diff
@@ -0,0 +1,86 @@
+angular.module('bhima.services')
+  .service('FillFormService', FillFormService);
+
+FillFormService.$inject = ['PrototypeApiService'];
+
+/**
+ * @class FillFormService
+ * @extends PrototypeApiService
+ *
+ * @description
+ * Encapsulates common requests to the /fill_form/ URL.
+ */
+function FillFormService(Api) {
+  const service = new Api('/fill_form/');
+
+  service.formatData = formatData;
+  service.restoreImage = restoreImage;
+  service.formatConstraint = formatConstraint;
+
+  function formatData(form, data) {
+    const dataSurveyForm = {};
+    // Ici l'on verifie les occurences des donnees existantes
+    // S'il sont multiples afin de les affecter dans des tableaux
+    form.forEach(f => {
+      data.forEach(d => {
+        if ((f.id === d.survey_form_id) && (f.type === '4')) {
+          dataSurveyForm[f.name] = [];
+        }
+      });
+    });
+
+    form.forEach(f => {
+      data.forEach(d => {
+        // Pour les identifiants des listes des choix,
+        // il est primordiale de le transormer en Integer
+        if ((f.id === d.survey_form_id) && (f.type === '4')) {
+          // Multiple choice list management
+          dataSurveyForm[f.name].push(parseInt(d.value, 10));
+        } else if ((f.id === d.survey_form_id) && (f.type !== '4') && (f.type !== '7')) {
+          // Single choice list management
+          d.value = (f.type === '1') ? parseFloat(d.value) : d.value;
+          d.value = (f.type === '3') ? parseInt(d.value, 10) : d.value;
+
+          // format date
+          d.value = (f.type === '6') ? new Date(d.value) : d.value;
```

Comment:
You are writing something like `d.value = (f.type === x) ? onething() : d.value` over and over again.  Let's  extract this code into a separate parser function that uses a [`switch`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) statement.  Something like this:

```js
d.value = parseDataValue(d.value, f.type);


// somewhere 
function parseDataValue(value, type) {
  switch(type) {
    case '1':
      return parseInt(value, 10);
    case '4':
      return new Date(value);
    default:
      return value;
  }
}
```

If you make this a function _and_ you implement the constants suggestion above, you will get a really nice looking function:
```js
function parseDataValue(value, type) {
  switch(type) {
    case QUESTION_INTEGER_TYPE:
      return parseInt(value, 10);
    case QUESTION_DATE_TYPE:
      return new Date(value);
    // etc...
    default:
      return value;
  }
}
```
My syntax for the switch function may be wrong, be sure to test it out.